pallets/xcmp-handler/rpc/src/lib.rs

```diff
@@ -0,0 +1,120 @@
+use sp_runtime::{generic::BlockId, traits::Block as BlockT, AccountId32};
+use std::{fmt::Debug, sync::Arc};
+
+/// An RPC endpoint to provide information about xcmp.
+#[rpc(client, server)]
+pub trait XcmpHandlerApi<Block, Balance> {
+	#[method(name = "xcmpHandler_crossChainAccount")]
+	fn cross_chain_account(&self, account: AccountId32) -> RpcResult<AccountId32>;
+	#[method(name = "xcmpHandler_fees")]
+	fn fees(&self, encoded_xt: Bytes) -> RpcResult<u64>;
+}
+
+/// An implementation of XCMP-specific RPC methods on full client.
+pub struct XcmpHandler<C, B> {
+	client: Arc<C>,
+	_marker: std::marker::PhantomData<B>,
+}
+
+impl<C, B> XcmpHandler<C, B> {
+	/// Create new `XcmpHandlerUtility` with the given reference to the client.
+	pub fn new(client: Arc<C>) -> Self {
+		Self { client, _marker: Default::default() }
+	}
+}
+
+/// Error type of this RPC api.
+pub enum Error {
+	/// The call to runtime failed.
+	RuntimeError,
+}
+
+impl From<Error> for i32 {
+	fn from(e: Error) -> i32 {
+		match e {
+			Error::RuntimeError => 1,
+		}
+	}
+}
+
+#[async_trait]
+impl<C, Block, Balance> XcmpHandlerApiServer<<Block as BlockT>::Hash, Balance>
+	for XcmpHandler<C, Block>
+where
+	Block: BlockT,
+	Balance: Codec + Copy + TryInto<u64> + Debug,
+	C: Send + Sync + 'static + ProvideRuntimeApi<Block> + HeaderBackend<Block>,
+	C::Api: XcmpHandlerRuntimeApi<Block, Balance>,
+{
+	fn cross_chain_account(&self, account_id: AccountId32) -> RpcResult<AccountId32> {
+		let api = self.client.runtime_api();
+		let at = BlockId::hash(self.client.info().best_hash);
+		let runtime_api_result = api.cross_chain_account(&at, account_id);
+		let mapped_err = |message| -> JsonRpseeError {
+			JsonRpseeError::Call(CallError::Custom(ErrorObject::owned(
+				Error::RuntimeError.into(),
+				"Unable to get cross chain AccountId",
+				Some(message),
+			)))
+		};
+		runtime_api_result
+			.map_err(|e| mapped_err(format!("{:?}", e)))
+			.map(|r| r.map_err(|e| mapped_err(String::from_utf8(e).unwrap_or(String::default()))))?
+	}
+
+	fn fees(&self, encoded_xt: Bytes) -> RpcResult<u64> {
+		let api = self.client.runtime_api();
+		let at = BlockId::hash(self.client.info().best_hash);
+		let runtime_api_result = api.fees(&at, encoded_xt).map_err(|e| {
+			CallError::Custom(ErrorObject::owned(
+				Error::RuntimeError.into(),
+				"Unable to get fees",
+				Some(e.to_string()),
+			))
+		})?;
+		runtime_api_result
+			.map(|v| {
+				v.try_into().map_err(|_| {
+					JsonRpseeError::Call(CallError::Custom(ErrorObject::owned(
+						Error::RuntimeError.into(),
+						"RPC value doesn't fit in u64 representation",
+						Some(format!("RPC value cannot be translated into u64 representation")),
+					)))
+				})
+			})
+			.map_err(|_| {
+				JsonRpseeError::Call(CallError::Custom(ErrorObject::owned(
+					Error::RuntimeError.into(),
+					"RPC value doesn't fit in u64 representation",
+					Some(format!("RPC value cannot be translated into u64 representation")),
+				)))
+			})?
```

Comment:
This is bad

Reply:
agreed but we can circle back tomorrow. Just want to get mangata unblocked